src/components/CodeExamples.astro

```diff
@@ -0,0 +1,79 @@
+---
+import { Code } from "astro:components";
+import ChevronRightIcon from "./icons/ChevronRightIcon.astro";
+import ExternalIcon from "./icons/ExternalIcon.astro";
+import grainLang from "../../grain-language-server/editor-extensions/vscode/syntaxes/grain.json";
+import githubDarkModifiedTheme from "../../themes/github-dark-modified.json";
+import lzString from "lz-string";
+import { codeExamples } from "../examples";
+
+const theme = githubDarkModifiedTheme as any;
+const lang = grainLang as any;
+---
+
+<script>
+  const mod = (n: number, k: number) => (n + k) % k;
+
+  const examples = document.getElementById("code-samples")!;
+  const selectedExample = document.getElementById("selected-example")!;
+  const playgroundLink = document.getElementById("playground-link")!;
+
+  const numExamples = examples.childElementCount;
+
+  let currIndex = 0;
+  function changeView(delta: number) {
+    const nextIndex = mod(currIndex + delta, numExamples);
+    examples.children[currIndex].classList.add("hidden");
+    examples.children[nextIndex].classList.remove("hidden");
+    currIndex = nextIndex;
+
+    const compressedCode = examples.children[currIndex].getAttribute("data-compressed-code");
+    playgroundLink.setAttribute("href", `/try?code=${compressedCode}`)
+
+    selectedExample.innerHTML = examples.children[currIndex].getAttribute("data-name")!;
+  }
+
+  document.getElementById("prev")!.addEventListener("click", () => changeView(-1));
+  document.getElementById("next")!.addEventListener("click", () => changeView(+1));
+
+  changeView(0);
+</script>
+
+<div class="flex flex-col relative h-full w-full">
```

Comment:
In light mode this stays dark, I found the contrast between the light background and this to be way to much making it really hard to read. (I noticed we might have the same issue on codeblocks in the docs as well), I don't know the best fix but I think if we just made the purple a bit more faded in lightmode (maybe we head more towards a lavender) it would help a lot.

Reply:
Should we just use a light theme in light mode altogether? cc @ospencer @phated

Reply:
Where is this?